### PR TITLE
Add Searx Search tool

### LINKED ISSUE #2: searx Agent support
Thanks for publishing this, I started to read the codebase for Marvin and langchain a few days ago and was in desperate need of something simpler.

If you take requests for agents I would like to kindly ask for searchx support. It can be self hosted, or given that searx is a python package even just imported to avoid the network call! By doing that we could have access to a lot of plugins like Github and Wolfram without writing all those agents.

### PR DESCRIPTION
**Pull request by Zetaphor**
This PR adds support for searching from Searx instances. Set the `SEARX_INSTANCE_URL` environment variable to the URL of your instance.

Note that this relies on the JSON output format being enabled in the instance, which it is not by default.

This can be enabled in the settings.yml by adding the JSON key:

https://github.com/searxng/searxng/blob/934249dd05142cde3461c8c4aae2c6d5804b0409/searx/settings.yml#L63

This resolves #2.

### CODE REVIEW

File: llm_agents/searx.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import os
+from typing import Any
+from llm_agents.tools.base import ToolInterface
+import requests
+from urllib.parse import urlencode
```
Comment (mpaepper) [inline llm_agents/searx.py]: Seems unused

File: llm_agents/searx.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+def _searx_search_results(params) -> list[dict[str, Any]]:
+    search_params = {
+        "q": params['q'],
+        "safesearch": 0,
+        "format": "json",
+    }
+
+    if params['safesearch']:
+        search_params['safe'] = 1
```
Comment (mpaepper) [inline llm_agents/searx.py]: Should be 'safesearch' or 'safe' (above is safesearch)

File: llm_agents/searx.py (hunk 1 above)
Comment (Zetaphor) [reply llm_agents/searx.py]: Correct, this was an artifact from testing with POST. Resolved in latest commit

File: llm_agents/searx.py (hunk 2 above)
Comment (Zetaphor) [reply llm_agents/searx.py]: Definitely a typo, it should be `safesearch` according to the API docs

**Comment by Zetaphor** [comment]
Thanks for the review, can you please take another look when you have a moment?

**Comment by mpaepper** [comment]
Please also make it available in the __init__. py

**Comment by Zetaphor** [comment]
All set!

**Pull request merged**